### ISSUE TITLE
problem with some file names

### ISSUE THREAD

**Issue by GianlucaApollaro** [nvaccess/osara#38]
Hi, I encountered an issue concerning file names Using latest osara in both latest reaper 4 and reaper 5. If you insert any media item that has an accented character in its name, the whole reaper crashes, returning a runtime error. It doesn't happen with rea-access or when osara is not loaded. It started happening a couple of days ago

**Comment by jcsteh** [comment]
I can't reproduce this. Are you definitely running the latest build of 
OSARA? Can you give me a file name that will trigger this? What do you 
have your Windows language set to: Italian or English?

**Comment by jcsteh** [comment]
I just tried this with two files, one named t•e.wav and the other named Michael Bublé.wav. Both worked as expected when inserting and then moving to the item. What screen reader are you using?

**Comment by GianlucaApollaro** [comment]
It happens with both latest jaws and latest NVDA. I'm using everything 
64 bit. The files I used were named simply è.wav and é.wav

Il 29/08/2015 01:48, James Teh ha scritto:

**Comment by jcsteh** [comment]
I just tested this with a file called è.wav on REAPER 64 bit and everything worked as expected. Are you absolutely certain this is the latest build of OSARA; i.e. including the new shortcut help mode and copy/cut/paste stuff?

Also, as I asked in a previous comment, what is your Windows language set to: English or Italian? I need as much detail as possible: Windows version, any error messages that appear, any system language settings, etc.

@ScottChesworth, would you mind testing this to see if you can reproduce? Simplest way to create such a file is to rename something and copy/paste one of the wav file names mentioned above.

**Comment by GianlucaApollaro** [comment]
Hi,
I'm positive that the build of osara is the latest one because I worked 
on the keymap to add the input help key. My windows is windows 7 64 bit 
in italian; I don't know if this matters but I'm using a portable copy 
of reaper. The exact error is:

Microsoft Visual C++ Runtime Library
---------------------------
Runtime Error!

This application has requested the Runtime to terminate it in an unusual 
way.
Please contact the application's support team for more information.

I'll have other italian users test and report back

Il 29/08/2015 11:59, James Teh ha scritto:

**Comment by jcsteh** [comment]
It shouldn't make a difference, but note that there is a build *after* 
the shortcut help build which includes cut/copy/paste reporting. I only 
released that yesterday.

**Comment by MatejGolian** [comment]
I've done some tests and I must say this is a quite interesting issue.
I'm running Windows 7 Ultimate X64 and I have the 64-bit version of
Reaper installed. My system language is set to Slovak. In my tests
I've used some Slovak and Swedish special characters in the file
names.
I got no errors when I used the "Media file" dialogue from the Insert
menu, but using the SWS command "Insert media file from clipboard"
procuced different results.

Here is what I got:
á.wav - error (see below)
å.wav - no error, but no file got inserted
ä.wav - error
č.wav - no error, but no file got inserted
é.wav - error
è.wav - error
ö.wav - error
ť.wav - no error, but no file got inserted
ž.wav - no error, but no file got inserted

This is the error message I got each time when inserting a
"problematic" file directly from the clipboard. Note that the error
appears when you insert a file with a problematic file name and press
the up or down arrow. After that Reaper completely stops working:

"Runtime Error!
Program: C:\Program Files\REAPER (x64)\reaper.exe
This application has requested the Runtime to terminate it in an
unusual way.
Please contact the application's support team for more information.

OK"

It doesn't seem to be a problem if the path contains one of these
special characters - the file names seem to be the only issue.
I'm certain that I have the latest build of Osara. My Reaper version
is 5.0.1 and my SWS version is 2.8.0 (the latest one).

Hope that helped.

2015-08-29 11:59 GMT+02:00, James Teh <notifications@github.com>:

**Comment by GianlucaApollaro** [comment]
I redownloaded latest build and tested again, the error occurs both when 
using the sws action and inporting the file from the insert media file 
menu. I am also running Windows 7 Ultimate X64 bit version. Sws version 
is 2.8.0 (, but again, it happens also when using the menu to insert a file.
Hth.


Il 29/08/2015 13:35, MatejGolian ha scritto:

**Comment by MatejGolian** [comment]
That's interesting, I have no problem using the 'menu' way.

2015-08-29 13:53 GMT+02:00, GianlucaApollaro <notifications@github.com>:

**Comment by MatejGolian** [comment]
Update: It keeps on happening no matter what display language I switch
to - English, Swedish or Slovak - makes no difference. I don't use any
special characters in my file names (at least not when working with
Reaper), so to me this is not much of an issue (especially if it turns
out that Osara doesn't have anything to do with it), but it would be
great if more people could test this. Could be that it's "only" our
systems that have been compromised. Unfortunately, I don't have a
fresh install of Windows to make further tests on at the moment.

2015-08-29 14:00 GMT+02:00, Matej Golian <matej.golian@gmail.com>:

**Comment by jcsteh** [comment]
I was previously using Windows 8.1 and now 10. It sounds like this might be a Windows 7 specific issue, since you're both running that. Unfortunately, my virtual test machines are all broken right now, so it'll probably be a while before I can look into this.

**Comment by jcsteh** [comment]
I just tested this in Windows 7 x64 with the Insert File dialog and I'm still not seeing this problem with a file called é.wav. However, I do see it with SWS Insert from Clipboard.

Can you please provide *exact* steps to reproduce the Insert File issue? That is, you hit the insert key, then what exactly do you do? Do you type a file name, shift+tab to the list and navigate to find a file, etc.?

**Comment by jcsteh** [comment]
I've tried to hack around this now, but I'd still be keen to see exact steps for reproducing this with the Insert File dialog. I still can't reproduce it that way.

IMO, this is a bug in REAPER and/or SWS, but whatever. :)

**Comment by GianlucaApollaro** [comment]
Hi, now it works in both ways, inserting from the menu and inserting 
from the clipboard. The steps were the one you described, insert menu, 
media file and select a file from a folder. What made me think it was 
not a reaper/sws bug was the fact that I got the crash also when opening 
a project containing one of those files that opened just fine with same 
reaper/sws a month ago. And olso with same reaper4/sws, with rea-access 
loaded.

Il 03/09/2015 07:48, James Teh ha scritto:

**Comment by jcsteh** [comment]
I did make a change back in April to properly support Unicode text. The 
crash wouldn't have happened before then, but there would also have been 
cases when you got very odd characters. What I don't understand is why 
the Insert File dialog results in invalid UTF-8 on your system but not 
with several machines I tested. Insert from clipboard is different; it 
looks like that's buggy regardless of what system.
